src/graph/sampler.cc

```diff
@@ -1592,17 +1629,31 @@ class WeightedEdgeSamplerObject: public EdgeSamplerObject {
   }
 
   void Fetch(DGLRetValue* rv) {
+    const int64_t num_workers = std::min(num_workers_, max_batch_id_ - curr_batch_id_);
     // generate subgraphs.
-    std::vector<SubgraphRef> positive_subgs(num_workers_);
-    std::vector<SubgraphRef> negative_subgs(num_workers_);
+    std::vector<SubgraphRef> positive_subgs(num_workers);
+    std::vector<SubgraphRef> negative_subgs(num_workers);
+
 #pragma omp parallel for
-    for (int i = 0; i < num_workers_; i++) {
+    for (int i = 0; i < num_workers; i++) {
       const dgl_id_t *seed_edge_ids = static_cast<const dgl_id_t *>(seed_edges_->data);
-      std::vector<int64_t> edge_ids(batch_size_);
+      std::vector<size_t> edge_ids(batch_size_);
 
-      for (int i = 0; i < batch_size_; ++i) {
-        int64_t edge_id = edge_selector_->Sample();
-        edge_ids[i] = seed_edge_ids[edge_id];
+#pragma omp critical
```

Comment:
i think you need to move this line after line 1644.

Reply:
I see.